sentry/src/main/java/io/sentry/TraceContext.java

```diff
@@ -70,142 +110,37 @@ public final class TraceContext implements JsonUnknown, JsonSerializable {
     return environment;
   }
 
-  public @Nullable TraceContextUser getUser() {
-    return user;
+  public @Nullable String getUserId() {
+    return userId;
+  }
+
+  public @Nullable String getUserSegment() {
+    return userSegment;
   }
 
   public @Nullable String getTransaction() {
     return transaction;
   }
 
+  public @Nullable String getSampleRate() {
+    return sampleRate;
+  }
+
   public @NotNull Baggage toBaggage(@NotNull ILogger logger) {
     Baggage baggage = new Baggage(logger);
 
     baggage.setTraceId(traceId.toString());
     baggage.setPublicKey(publicKey);
+    baggage.setSampleRate(sampleRate);
     baggage.setRelease(release);
     baggage.setEnvironment(environment);
     baggage.setTransaction(transaction);
-
-    final TraceContextUser user = this.user;
-    if (user != null) {
-      baggage.setUserId(user.id);
-      baggage.setUserSegment(user.segment);
-    }
+    baggage.setUserId(userId);
+    baggage.setUserSegment(userSegment);
 
     return baggage;
   }
 
-  public static final class TraceContextUser implements JsonUnknown, JsonSerializable {
```

Comment:
Would that be a breaking change if people is upgrading SDKs from X to Y? how does the ser/deser behave if you're reading a JSON with the older structure.

Reply:
Yes, updated the code to parse from legacy JSON